simplesat/package.py

```diff
@@ -52,9 +56,33 @@ def _from_pretty_string(cls, s):
         return parser.parse_to_package(s)
 
     def __init__(self, name, version, install_requires=None):
+        """ Return a new PackageMetdata object.
+
+        Parameters
+        ----------
+        name : str
+            The name of the Python distribution, e.g. "numpy"
+        version : EnpkgVersion
+            An EnpkgVersion object describing the version of this package.
+        install_requires : tuple(tuple(str, tuple(tuple(str))))
+            A tuple of tuples mapping distribution names to disjunctions of
+            conjunctions of version constraints.
+
+            For example, a consider a package that depends on the following:
+                - nose
+                - six (> 1.2, <= 1.2.3), or >= 1.2.5-2
+                    Written as intervals, (1.2, 1.2.3] or [1.2.5-2, \infty)
+                - MKL >= 10.1, < 11
+
+            The constraint tuple representing this would be:
+                (("MKL", ((">= 10.1", "< 11"),)),
+                 ("nose", (("",),)),
+                 ("six", (("> 1.2", "<= 1.2.3"), (">= 1.2.5-2",)))
+
+        """
         self._name = name
         self._version = version
-        self._install_requires = install_requires or tuple()
+        self._install_requires = tuple(sorted(install_requires or ()))
```

Comment:
I'm not sure I follow. Can you give an example that demonstrates which part you're thinking needs to be sorted and which part doesn't? Right now, the code sorts based on the names of the reverse-dependencies, but not the constraints associated with them.

For example, the way it currently is we get the following behavior.

```
>>> sorted("><")
['<', '>']

>>> pkg = ppsp.parse("numpy 1.8.0-2; depends (zope > 2.0, bokeh, zope < 1.0)")
>>> pkg['install_requires']
(('bokeh', (('',),)), ('zope', (('> 2.0', '< 1.0'),)))
```

So the order of the package names gets sorted, but the order of the constraint
strings does not. This sounds like what you were suggesting, but can you
confirm?

I just cleaned this up a little bit as well.